### PR TITLE
Refine PR Builder Code

### PR DESCRIPTION
**Pull request by saadmk11**
This PR Will work on improving the code for PR builder by updating the code suggested in code reviews

### CODE REVIEW

File: readthedocs/core/views/hooks.py [hunk 1]
```diff
@@ -105,30 +105,23 @@ def get_or_create_external_version(project, identifier, verbose_name):
...
-        manager=EXTERNAL
-    ).filter(verbose_name=verbose_name).first()
+    external_version, created = project.versions.get_or_create(
+        verbose_name=verbose_name,
+        type=EXTERNAL,
+        defaults={'identifier': identifier, 'active': True},
+    )
 
-    if external_version:
+    if not created:
```
Comment (ericholscher) [inline readthedocs/core/views/hooks.py]: I'd switch this logic around, and do `if created`. It just reads a bit cleaner.

File: readthedocs/core/views/hooks.py [hunk 2]
```diff
@@ -105,30 +105,23 @@ def get_or_create_external_version(project, identifier, verbose_name):
...
         # identifier will change if there is a new commit to the Pull/Merge Request
         if external_version.identifier != identifier:
             external_version.identifier = identifier
             external_version.save()
```
Comment (ericholscher) [inline readthedocs/core/views/hooks.py]: We should probably log in this case too (eg. updated external version)

File: readthedocs/templates/builds/build_detail.html [hunk 3]
```diff
@@ -91,7 +91,11 @@
         {% endif %}
       </span>
       <span class="build-commit" data-bind="visible: commit">
+        {% if build.get_commit_url %}
           (<a data-bind="attr: {href: commit_url}"><span data-bind="text: commit">{{ build.commit }}</span></a>)
+        {% else %}
+            (<span data-bind="text: commit">{{ build.commit }}</span>)
```
Comment (ericholscher) [inline readthedocs/templates/builds/build_detail.html]: Does this break the JS code that tries to update `commit_url` if it doesn't exist? I don't know how knockout.js handles this, but we might also need to check for it on that side.

Comment (saadmk11) [reply readthedocs/templates/builds/build_detail.html]: Okay I'll check this :)

Comment (saadmk11) [reply readthedocs/templates/builds/build_detail.html]: I have checked. there is no error on the browsers console window. and `commit_url` does always exist. but set to `None` when conditions are not met.

File: readthedocs/projects/tasks.py [hunk 4]
```diff
@@ -578,18 +578,21 @@ def run_build(self, docker, record):
 
         if self.build_env.failed:
             self.send_notifications(self.version.pk, self.build['id'])
-            # send build failure status to git Status API
+            # if the build failed, send build failure status to git Status API
+            # to show status report on the providers pull/merge request UI.
```
Comment (ericholscher) [inline readthedocs/projects/tasks.py]: This is still just explaining what the code does. We can just remove these comments, we don't need to explain anything here.

File: readthedocs/core/views/hooks.py [hunk 5]
```diff
@@ -105,30 +105,29 @@ def get_or_create_external_version(project, identifier, verbose_name):
...
+        verbose_name=verbose_name,
+        type=EXTERNAL,
+        defaults={'identifier': identifier, 'active': True},
+    )
 
-    if external_version:
+    if created:
+        log.info(
+            '(Create External Version) Added Version: [%s] ', ' '.join(
+                external_version.slug
```
Comment (ericholscher) [inline readthedocs/core/views/hooks.py]: Why are we calling `join` on a slug?

Comment (saadmk11) [reply readthedocs/core/views/hooks.py]: Opps! I was doing something else forgot to remove it sorry.

**Pull request merged**